### PR TITLE
POM cleanup

### PR DESCRIPTION
**Pull request by jglick**
Various problems in the POM.

Not tested “live” since I barely understand what this plugin does (not needed by anything Pipeline, anyway) and am not sure where I would see it being used.

@reviewbybees

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -4,13 +4,14 @@
   <parent>
     <groupId>org.jenkins-ci.plugins</groupId>
     <artifactId>plugin</artifactId>
-    <version>2.19</version>
+    <version>2.33</version>
```
Comment (jglick) [inline pom.xml]: Routine update.

File: pom.xml [hunk 2]
```diff
@@ -4,13 +4,14 @@
   <parent>
     <groupId>org.jenkins-ci.plugins</groupId>
     <artifactId>plugin</artifactId>
-    <version>2.19</version>
+    <version>2.33</version>
+    <relativePath />
```
Comment (jglick) [inline pom.xml]: Good form, since `../pom.xml` is not meaningful.

File: pom.xml [hunk 3]
```diff
@@ -4,13 +4,14 @@
...
-    <version>2.19</version>
+    <version>2.33</version>
+    <relativePath />
   </parent>
 
   <properties>
     <jenkins.version>1.642.3</jenkins.version>
     <java.level>7</java.level>
-    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <findbugs.failOnError>false</findbugs.failOnError> <!-- TODO fix the warnings instead -->
```
Comment (jglick) [inline pom.xml]: 2.x parent handles this.

File: pom.xml [hunk 4]
```diff
@@ -44,22 +45,9 @@
   <scm>
     <connection>scm:git:git://github.com/jenkinsci/token-macro-plugin.git</connection>
     <developerConnection>scm:git:git@github.com:jenkinsci/token-macro-plugin.git</developerConnection>
-    <tag>token-macro-2.1</tag>
+    <tag>HEAD</tag>
```
Comment (jglick) [inline pom.xml]: Seems like you had some kind of snafu with `maven-release-plugin`?

File: pom.xml [hunk 5]
```diff
@@ -70,6 +58,25 @@
       <groupId>org.parboiled</groupId>
       <artifactId>parboiled-java</artifactId>
       <version>1.1.7</version>
+      <exclusions>
+        <!-- Jenkins core bundles ASM 4, so if this really requires ASM 5 and is used at runtime, you are SOL. -->
```
Comment (jglick) [inline pom.xml]: In other words, this plugin was bundling ASM 5 (actually several copies…) which made it look like other (library) classes bundled in the plugin were linking against that, whereas in fact the baseline version of Jenkins core bundles ASM 4, which takes precedence. Current versions bundle 5.

File: pom.xml [hunk 6]
```diff
@@ -81,14 +88,18 @@
...
-    <dependency>
-      <groupId>org.jenkins-ci.plugins.workflow</groupId>
-      <artifactId>workflow-job</artifactId>
-      <version>2.0</version>
+      <exclusions>
+        <exclusion> <!-- Note that json-smart *also* bundles ASM 5 inline, which we cannot do anything about. Bad form! -->
+          <groupId>org.ow2.asm</groupId>
+          <artifactId>asm</artifactId>
+        </exclusion>
+        <exclusion> <!-- inlines it anyway, no need to bundle a duplicate copy (but note that licenses.xml will be incomplete) -->
```
Comment (jglick) [inline pom.xml]: Evil usage of `org.apache.felix:maven-bundle-plugin`.

File: pom.xml [hunk 7]
```diff
@@ -81,14 +88,18 @@
       <groupId>com.jayway.jsonpath</groupId>
       <artifactId>json-path</artifactId>
       <version>2.1.0</version>
-    </dependency>
-    <dependency>
-      <groupId>org.jenkins-ci.plugins.workflow</groupId>
-      <artifactId>workflow-job</artifactId>
```
Comment (jglick) [inline pom.xml]: Really should not have had this as a `compile`-scoped dep (i.e., runtime plugin dependency).

File: pom.xml [hunk 8]
```diff
@@ -126,13 +143,6 @@
         </exclusion>
       </exclusions>
     </dependency>
-    <dependency>
-      <groupId>org.jenkins-ci.main</groupId>
-      <artifactId>jenkins-war</artifactId>
-      <type>war</type>
```
Comment (jglick) [inline pom.xml]: Why was this here??

File: src/main/java/org/jenkinsci/plugins/tokenmacro/impl/BuildStatusMacro.java [hunk 9]
```diff
@@ -43,7 +40,8 @@ public String evaluate(Run<?,?> run, FilePath workspace, TaskListener listener,
             throws MacroEvaluationException, IOException, InterruptedException {
 
         // In the case of pipeline jobs, if the status hasn't been set to a non-null value, then it is considered "success"
-        if(run instanceof WorkflowRun && null == run.getResult()) {
+        if (!(run instanceof AbstractBuild) && null == run.getResult()) {
+            // TODO this makes little sense. Why not just use Building as below?
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/tokenmacro/impl/BuildStatusMacro.java]: TBD, but anyway suffices to check for non-`AbstractBuild`s.

Comment (slide) [reply src/main/java/org/jenkinsci/plugins/tokenmacro/impl/BuildStatusMacro.java]: This was a specific request so that email-ext stuff would work during a build.

File: pom.xml (hunk 8 above)
Comment (slide) [reply pom.xml]: Good question

File: src/main/java/org/jenkinsci/plugins/tokenmacro/impl/BuildStatusMacro.java (hunk 9 above)
Comment (jglick) [reply src/main/java/org/jenkinsci/plugins/tokenmacro/impl/BuildStatusMacro.java]: Right, but would that not be covered by the check for a null result ~ `Building` below this already?

Comment (slide) [reply src/main/java/org/jenkinsci/plugins/tokenmacro/impl/BuildStatusMacro.java]: They didn't want 'Building' they wanted a Success, but the job hasn't completed yet. So, it was an intermediate 'Success' they were looking for. See https://issues.jenkins-ci.org/browse/JENKINS-44322

**Comment by recena** [comment]
@slide Hello Could you cut a new release by including this PR? Thanks.

**Comment by slide** [comment]
Yes, I can do that.

**Comment by recena** [comment]
@slide I really appreciate it.

**Comment by slide** [comment]
Release was done this morning. Sorry it took so long!
